server/src/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1924,6 +1935,23 @@ public Volume migrateVolume(MigrateVolumeCmd cmd) {
                                 throw new InvalidParameterValueException("Cannot migrate ROOT volume of a stopped VM to a storage pool in a different VMware datacenter");
                             }
                         }
+
```

Comment:
What about extracting lines 1939-1954 to a method? This can really help us to work on that code and the unit tests it deserves.